### PR TITLE
demo env enable dns

### PR DESCRIPTION
**Pull request by naikajah**

### CODE REVIEW

File: src/uk/gov/hmcts/pipeline/AKSSubscriptions.groovy [hunk 1]
```diff
@@ -26,5 +27,8 @@ class AKSSubscriptions {
     def prodName = steps.env.AKS_PROD_SUBSCRIPTION_NAME ?: 'DCD-CFTAPPS-PROD'
     def prodKvName = steps.env.PROD_AKS_KEY_VAULT ?: 'cft-apps-prod'
     prod = new AKSSubscription(steps, prodName, prodKvName, 'prod', true)
+
+    def demoName = steps.env.AKS_DEMO_SUBSCRIPTION_NAME ?: 'DCD-CFTAPPS-DEMO'
+    prod = new AKSSubscription(steps, demoName, 'cftapps-demo', 'demo', true)
```
Comment (adusumillipraveen) [inline src/uk/gov/hmcts/pipeline/AKSSubscriptions.groovy]:
```suggestion
    demo = new AKSSubscription(steps, demoName, 'cftapps-demo', 'demo', true)
```

File: src/uk/gov/hmcts/pipeline/AKSSubscriptions.groovy [hunk 2]
```diff
@@ -26,5 +27,8 @@ class AKSSubscriptions {
     def prodName = steps.env.AKS_PROD_SUBSCRIPTION_NAME ?: 'DCD-CFTAPPS-PROD'
     def prodKvName = steps.env.PROD_AKS_KEY_VAULT ?: 'cft-apps-prod'
     prod = new AKSSubscription(steps, prodName, prodKvName, 'prod', true)
+
+    def demoName = steps.env.AKS_DEMO_SUBSCRIPTION_NAME ?: 'DCD-CFTAPPS-DEMO'
+    demo = new AKSSubscription(steps, demoName, 'cftapps-demo', 'demo', true)
```
Comment (adusumillipraveen) [inline src/uk/gov/hmcts/pipeline/AKSSubscriptions.groovy]: Do it same as prod ? and get it from env variable? like https://github.com/hmcts/cnp-jenkins-library/pull/596/files#diff-6e016a19ecdb9ae05ca73906b312360cR29

Comment (adusumillipraveen) [reply src/uk/gov/hmcts/pipeline/AKSSubscriptions.groovy]: never mind, ignore it.

File: vars/autoDeployEnvironment.groovy [hunk 3]
```diff
@@ -20,8 +20,8 @@ def call() {
     ],
     demo: [
       environmentName: environment.demoName,
-      subscriptionName: subscription.nonProdName,
-      aksSubscription: null,
+      subscriptionName: subscription.demoName,
+      aksSubscription: aksSubscriptions.demoName,
       aksInfraRgName: null
```
Comment (timja) [inline vars/autoDeployEnvironment.groovy]:
```suggestion
```

**Pull request merged**